web3/utils/encoding.py

```diff
@@ -175,7 +175,10 @@ def to_decimal(value=None, hexstr=None, text=None):
             ))
             return to_decimal(hexstr=value)
         else:
-            return int(value)
+            try:
+                return int(value)
+            except ValueError:
+                return to_decimal(hexstr=to_hex(value))
```

Comment:
These strike me as something that should have a corresponding test case for whatever failure they are handling?

Reply:
done